### PR TITLE
br: fix unstable unit test

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31356

Problem Summary:  https://github.com/pingcap/tidb/pull/31248/files roll back the fix of mockCluster conflict. This PR tries to fix it again...

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/client_test.go [hunk 1]
```diff
@@ -23,17 +23,15 @@ import (
 	"google.golang.org/grpc/keepalive"
 )
 
+var MC *mock.Cluster
```
Comment (kennytm) [inline br/pkg/restore/client_test.go:26]: this does not need to be public.

call the variable `mockCluster` or something descriptive.

File: br/pkg/restore/main_test.go [hunk 2]
```diff
@@ -28,5 +31,16 @@ func TestMain(m *testing.M) {
 		goleak.IgnoreTopFunction("go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop"),
 		goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),
 	}
-	goleak.VerifyTestMain(m, opts...)
+
+	MC, _ = mock.NewCluster()
+	MC.Start()
```
Comment (kennytm) [inline br/pkg/restore/main_test.go:36]: need to check if these two lines return error?

**Review by glorv — APPROVED** [review]

**Review by kennytm — APPROVED** [review]

**Pull request merged**